File: backend/app/schemas/curriculum.py
from pydantic import BaseModel


class LessonResponse(BaseModel):
    id: str
    title: str
    description: str
    template_name: str
    difficulty: str
    order: int


class ModuleResponse(BaseModel):
    id: str
    title: str
    description: str
    order: int
    lessons: list[LessonResponse]


class CurriculumResponse(BaseModel):
    modules: list[ModuleResponse]

File: frontend/src/types/curriculum.ts
export interface LessonResponse {
  id: string;
  title: string;
  description: string;
  template_name: string;
  template_id: string | null;
  difficulty: string;
  order: number;
}

export interface ModuleResponse {
  id: string;
  title: string;
  description: string;
  order: number;
  lessons: LessonResponse[];
}

export interface CurriculumResponse {
  modules: ModuleResponse[];
}
